src/Storages/MergeTree/MergeTreePartInfo.cpp

```diff
@@ -108,15 +109,24 @@ std::optional<MergeTreePartInfo> MergeTreePartInfo::tryParsePartName(
     /// Sanity check
     if (min_block_num > max_block_num)
         return std::nullopt;
-
+    /// Mutation number or/and _tryN suffix
     if (!in.eof())
     {
-        if (!checkChar('_', in)
-            || !tryReadIntText(mutation, in)
-            || !in.eof())
-        {
+        if (!checkChar('_', in))
             return std::nullopt;
+        if (tryReadIntText(mutation, in))
```

Comment:
Hm, actually it shouldn't be a part of a regular `MergeTreePartInfo::tryParsePartName` because it only makes sense for detached parts